public/js/results.js

```diff
@@ -2,74 +2,78 @@
 
 //create a file-specific context via a function
 (function(Piv, ElectionId) {
+    // snapshot versions (corresponds to ResultSnapshotController.php)
```

Comment:
Let's think about how to do this.  I like the idea of deduplicating this in the code.

However, I don't want the JS side to dynamically treat the server as the source of truth, because there are reasons that either the JS side or the server side could be ahead of the other.  In particular, there is the (less likely) case that CDNs or other caching could mean that client and server side are not rolled out at the same time.  In the (more likely) case, the export/import of snapshots mean that you could be exporting from one Pivot deployment and then importing in a different one that is either older or newer.

You can also imagine that we may be able to open snapshots when offline, as long as you have the JS page cached locally, since there is no necessary dependency on the server.